src/azure-cli-core/azure/cli/core/commands/__init__.py

```diff
@@ -518,6 +519,10 @@ def execute(self, args):
         self.cli_ctx.invocation.data['command_string'] = command
         telemetry.set_raw_command_name(command)
 
+        # record execution log when "az next" is installed
+        log_cmd_history(command, args)
+        clean_exception_history(command)
```

Comment:
Core should not be responsible for calling these 2 functions.

```suggestion
        prepare_for_next(command, args)
```

Reply:
Good suggestion!